src/libcore/iter.rs

```diff
@@ -280,6 +289,16 @@ pub trait Iterator {
     /// different sized integer, the `zip` function provides similar
     /// functionality.
     ///
+    /// # Undefined overflow
+    ///
+    /// The method does no guarding against overflows, so enumerating more than
+    /// `usize::MAX` elements is undefined.
```

Comment:
I believe we call it undefined overflow and don't guarantee anything about it other than it doesn't create `unsafe` things. It might panic, might give out seemingly-random numbers etc.